### ISSUE TITLE
LightningPose prediction in csv doesn't match with labels predicted on video

### ISSUE THREAD

**Issue by ReetKaur15** [danbider/lightning-pose#192]
Hi Team,

After training and inferencing LightningPose, I have the following CSV files in the output folder, 
1. Video1.csv
2. Video1_pca_singleview_error.csv
3. Video1_temporal_norm

I created a skeleton using LightningPose predicted CSV (Video1.csv) file and superimposed it over the Lightningpose predicted video (in the folder labeled_video/Video1_labeled.mp4). Upon looking at the new video (skeleton superimposed on video1_labeled), seems like predicted labels in the CSV file doesn't match with the labels marked on Video (please refer attached screenshot). How to get the values of keypoint coordinates marked on the predicted video (in the labeled_video/Video1_labeled) in csv file. 
<img width="65" alt="Capture11" src="https://github.com/user-attachments/assets/50e40cec-8062-4300-bf49-6c335b8b5a89">

Thanks :)

**Comment by themattinthehatt** [comment]
Huh that's interesting - can you share more about how you're plotting the skeleton on the frames? Does it seem like the skeleton is shifted in a particular direction spatially (up/down/etc) or in time (off by a frame or two)?

For reference, here is the function that runs inference and makes the labeled video: https://github.com/danbider/lightning-pose/blob/9ab15144e3d3d04c8d117f514fd1cdda9cc4b0e2/lightning_pose/utils/scripts.py#L669

**Comment by ReetKaur15** [comment]
@themattinthehatt , Thanks for your reply. I think you were right. 

Now, when i plotted only keypoints (rather than skeleton) in the CSV file on predicted video frames. It seems to be working fine. It must be an issue during skeleton creation or superimposing over plotted video. 

To create the skeleton, I extracted keypoints from the CSV file and wrote a script to combine pairs of keypoints, forming a meaningful skeleton by calculating joint and segment angles. I then superimposed the skeleton onto each video frame. However, there seems to be a time lag or a positional shift (up, down, right, left) during the superimposition.

All resolved now... Thanks once again. :) 

I have one more query, Does LightningPose training support vision transformers? I mean instead of using neural network architectures such as ResNet50, RestNet152, etc., can i train  LightningPose with transformers (ViT)?

**Comment by themattinthehatt** [comment]
Glad you got it fixed!

Yes Lighting Pose supports a range of different backbones - you can see a list here under the `model.backbone` entry: https://lightning-pose.readthedocs.io/en/latest/source/user_guide/config_file.html#model-training-parameters

Since you appear to be tracking humans you might want to try out `resnet50_human_jhmdb` or `resnet50_human_res_rle`, which have been pretrained on human datasets.

We also offer the option `vit_b_sam` which uses the pretrained Segment Anything 1.0 backbone. We have yet to do a thorough analysis of this transformer, but with our first attempts found that it didn't really perform better than a resnet50. 

Please let me know if you try some other backbones and how they work out for you!

**Comment by ReetKaur15** [comment]
Thanks, @themattinthehatt. Yes, I am estimating infant poses. Currently, I am using the RestNet152 network as a backbone. Is ResNet152, provided here, pretrained on a human dataset? 

Can i use vit_b_sam backbone, same as i am using ResNet152 by specifying its name in the config file ?

**Comment by themattinthehatt** [comment]
The ResNet-152 that you are using is pretrained on ImageNet; we only have ResNet-50s that are pretrained on human datasets. Might be worth trying one of those ResNet-50s just to see how it compares to the ResNet-152.

And yes, all you need to do is change the `backbone` field in the config file to `vit_b_sam` and you'll get the transformer backbone!

**Comment by ReetKaur15** [comment]
@themattinthehatt , Thanks alot for your feedback and information. I'll definitely let you know how it goes.. :)

Kind regards,
@ReetKaur15

**Comment by themattinthehatt** [comment]
No we haven't spent a lot of time exploring the different backbones. Here's a [paper](https://openaccess.thecvf.com/content/WACV2021/papers/Mathis_Pretraining_Boosts_Out-of-Domain_Robustness_for_Pose_Estimation_WACV_2021_paper.pdf) from the DLC people where they compare MobileNet, ResNet, and EfficientNet architectures, but that's the closest I to what you're looking for off the top of my head. 

If you want to dig into this yourself my suggestion would be to create an "out of distribution" (OOD) set of labeled data using subjects that are not in the training set. You can then train models with different backbones on the training set, and then test them on the OOD set and see how well the various backbones generalize for you.

How much labeled data do you have by the way? And how much variability is there in the background, and poses, etc.? If you don't have a ton of labeled data (low hundreds of frames) the ResNet-50 might be your best bet. If you have a lot of labeled images or there is a ton of variability the larger backbones (ResNet-152 or ViT) might be better. But performance can so often be dataset-specific, which is why trying a few options for yourself would ultimately be best in the long run.

**Comment by ReetKaur15** [comment]
Thanks, @themattinthehatt; 
we have around 8700 extracted frames; there is not much variability in the background, but we have a variety of poses in our dataset. We tested DLC with our dataset, it was performing better with the ResNet152, so opted same network for LP as well.
